WalletWasabi.Fluent/Models/Wallets/Address.cs

```diff
@@ -39,8 +39,7 @@ public Address(KeyManager keyManager, HdPubKey hdPubKey)
 
 	public void Hide()
 	{
-		// Code commented out due to https://github.com/zkSNACKs/WalletWasabi/issues/10177
-		// KeyManager.SetKeyState(KeyState.Locked, HdPubKey);
+		HdPubKey.SetLabel(LabelsArray.Empty, KeyManager);
```

Comment:
I'm sending the KM in, so it's written to file.

Reply:
This is risky IMO. With this, the client will be under the fake illusion that this address is unused and has never been never seen. So the next time it will give it back.
But the fact is once it was generated, and probably with a high chance shared with someone, it should be never used again.
That is why it got the `Locked` status, so the client knew that it should be ignored when the client generates the next receive key.

Reply:
Ah, you are right. On the next receive address we get back the same address, which is problematic.